sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetFilters.scala

```diff
@@ -40,6 +42,25 @@ private[sql] object ParquetFilters {
       createFilter(filter)
     }.reduceOption(FilterApi.and).map(FilterCompat.get)
   }
+    
+  case class SetInFilter[T <: Comparable[T]](
+    hSet: Set[T]) extends UserDefinedPredicate[T] with Serializable {
+
+    override def keep(value: T): Boolean = {
+      if (value == null) {
+        return false
+      }
+      return hSet.contains(value)
```

Comment:
Generally we try to avoid using `return` in Scala code. (You may find a detailed guide [here] [1].) For this method, we'd prefer:

```scala
value != null && hSet.contains(value)
```

BTW, why the name `hSet`? Maybe `valueSet` instead?

[1]: https://github.com/databricks/scala-style-guide